apps/lead/views.py

```diff
@@ -174,6 +175,13 @@ def get_queryset(self):
             ).filter(similarity__gt=0.3).order_by('-similarity')
         return leads
 
+    def destroy(self, *args, **kwargs):
+        lead = self.get_object()
+        result = super().destroy(*args, **kwargs)
+        # remove from index after the deletion is successful
+        remove_lead_from_index.delay(lead.id)
+        return result
+
```

Comment:
We will need to use signals as DRF views are not used anymore.

Reply:
Done.